napari/utils/colormaps/colormaps.py

```diff
@@ -111,6 +117,31 @@ def _low_discrepancy_image(image, seed=0.5, margin=1 / 256):
     return image_out
 
 
+def color_dict_to_colormap(color_dict):
+    """
+    Generate a color map based on the given color dictionary
+    Parameters
+    ----------
+    color_dict : dict of label and color string matches
+
+    Returns
+    -------
+    colormap : Colormap with provided control colors
+    custom_color : dict of label to color control point within colormap
+    """
+    custom_color = {0: 0.0}
+    colors = ['#000000'] + [
+        color_str for label, color_str in color_dict.items()
+    ]
+    colormap = Colormap(colors)
+    for label, color_str in color_dict.items():
+        color = Color(color_str)
+        for i in range(len(colormap.colors)):
+            if colormap.colors[i] == color:
+                custom_color[label] = colormap._controls[i]
```

Comment:
Can you explain why this needs to be a double for loop? Is it because we can't guarantee the order we get items from the dictionary? I wouldn't have thought two passes were needed.

Reply:
the outer loop is looping through the dictionary, and the inner loop is looping through colormap, I don't think there is a lookup function built-in colormap, so I have to do a lookup as a loop